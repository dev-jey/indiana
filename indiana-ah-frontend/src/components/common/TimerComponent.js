import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import ClockIcon from '../../assets/images/icons/clock.png';

const TimerComponent = ({ timeCount }) => (
  <Fragment>
    <span className="icon-holder">
      <img src={ClockIcon} alt="ClockIcon" className="icon-height" />
      <sub className="count">{timeCount}</sub>
    </span>
  </Fragment>
);

TimerComponent.propTypes = {
  timeCount: PropTypes.string.isRequired
};

export default TimerComponent;
