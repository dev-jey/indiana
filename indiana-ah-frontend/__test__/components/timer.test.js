import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TimerComponent from '../../src/components/common/TimerComponent';

describe('<BadgeComponent />', () => {
  const wrapper = shallow(<TimerComponent timeCount={'2 days ago'} />);
  it('should render a timer component', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper.find('span')).toHaveLength(1);
    expect(wrapper.find('img')).toHaveLength(1);
    expect(wrapper.find('sub')).toHaveLength(1);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<TimerComponent timeCount={'2 days ago'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
