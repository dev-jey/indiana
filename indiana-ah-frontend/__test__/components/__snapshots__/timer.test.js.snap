// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BadgeComponent /> renders correctly 1`] = `
<span
  className="icon-holder"
>
  <img
    alt="ClockIcon"
    className="icon-height"
    src="test-file-stub"
  />
  <sub
    className="count"
  >
    2 days ago
  </sub>
</span>
`;
